src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -154,10 +165,14 @@ def update(self, sample: Sample[QuantityT]) -> None:
         )
 
         # Update data
-        value: float = np.nan if sample.value is None else sample.value.base_value
+        if self.has_value(sample):
+            assert sample.value is not None
+            value = sample.value.base_value
+        else:
+            value = np.nan
```

Comment:
Or.... :exclamation::exclamation::exclamation: just not use the function in this context... The original code is pretty simple and understandable (and works the same, as if it is NaN we'll use `np.nan` anyway, or is it `np.nan` different from `float('nan')`?) without it too:

```py
        valueL = np.nan if sample.value is None else sample.value
        self._buffer[self.datetime_to_index(timestamp)] = value
        self._update_gaps(timestamp, prev_newest, not self.has_value(sample))
```

Reply:
I would like to leave it as it is required in both these places. So you have one location where you define what a missing value is and what not (e.g. if we want to change it to include `inf`s at some point.)

Actually I find it not so bad, even more readable. But yea, the assertion is annoying.